### ISSUE TITLE
Icon in button not centered correctly when using Bootstrap 4

### ISSUE THREAD

**Issue by Hypenate** [IgniteUI/igniteui-angular#762]
### Description  
When using bootstrap, a button with an icon is not centered correctly.

### Steps to reproduce  

1. Add igniteui-angular.css"
2. Add bootstrap.min.css" (v4)
3. In the HTML copy paste the igxButton from the website of igniteui:
```
<button igxButton="fab" igxButtonColor="#FBB13C" igxButtonBackground="#340068" igxRipple="#FBB13C">
  <igx-icon fontSet="material" name="edit"></igx-icon>
</button>
```

### Result  
The pencil is not placed in te center.
If you would remove the bootstrap.min.css the button is placed correctcly.
Placing the bootstrap.css file before or after the igniteui-angular.css makes no difference.

### Expected result  
The pencil in the center whilst using bootstrap 4.0

### Attachments  
Result:
![image](https://user-images.githubusercontent.com/7478681/37327769-ea0d0344-2697-11e8-8019-a0eff2aac23a.png)
Expected:
![image](https://user-images.githubusercontent.com/7478681/37327844-246cb3b8-2698-11e8-996d-2b018455a819.png)

**Comment by simeonoff** [comment]
I am unable to reproduce with the current master build. Used [this](https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-alpha.6/css/bootstrap.min.css) link and imported it before the all styles.

**Comment by Hypenate** [comment]
@simeonoff You are using Bootstrap v4.0.0-alpha.6, which is not te latest, try https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css

I have also included a project where you can see the issue.
[my_igniteui_project.zip](https://github.com/IgniteUI/igniteui-angular/files/1809918/my_igniteui_project.zip)

**Comment by simeonoff** [comment]
@Hypenate I've downloaded your project, ran it, and yes, I see what you mean. 

It's the igx-icon component causing the issue. The good news is the igx-icon component has been refactored as we were having other vertical alignment issues with it in other scenarios. 

Now, I ran your sample app with the current master and everything looks perfectly aligned.

I am closing this issue for now as it will be resolved with the next version release.
